test/jdk/java/lang/Thread/virtual/TracePinnedThreads.java

```diff
@@ -78,8 +79,66 @@ void testPinnedCausedBySynchronizedBlock() throws Exception {
     void testPinnedCausedByNativeMethod() throws Exception {
         System.loadLibrary("TracePinnedThreads");
         String output = run(() -> invokePark());
+        assertContains(output, "reason:NATIVE");
         assertContains(output, "(Native Method)");
-        assertDoesNotContain(output, "<== monitors");
+    }
+
+    /**
+     * Test parking in class initializer.
+     */
+    @Test
+    void testPinnedCausedByClassInitializer() throws Exception {
+        class C {
+            static {
+                park();
+            }
+        }
+        String output = run(C::new);
+        assertContains(output, "reason:NATIVE");
+        assertContains(output, "<clinit>");
+    }
+
+    /**
+     * Test contention writing to System.out when pinned.
+     */
+    @Test
+    void testContention() throws Exception {
```

Comment:
I think this test method  expects to have all the launched threads to print to System.out without hanging. If it hangs then we expect the test run to timeout with a jtreg timeout. Given the context of this PR in which this test method is being introduced, I think I can understand what it does. But maybe we should update the test method's comment to say that this method verifies that contention when writing to System.out when pinned doesn't cause any hangs?